enterprise/management/commands/unlink_enterprise_customer_learners.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+"""
+Django management command for unlink the learners and removing DSC.
+"""
+import csv
+import logging
+
+from django.core.management import BaseCommand
+
+from edx_django_utils.cache import get_cache_key, TieredCache
+from enterprise.models import EnterpriseCustomer, EnterpriseCustomerUser, \
+    PendingEnterpriseCustomerUser
+from django.contrib import auth
+
+from enterprise.utils import get_enterprise_customer_user
+
+User = auth.get_user_model()  # pylint: disable=invalid-name
+
+LOGGER = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Django management command for creating DataSharingConsent records
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--data-csv',
+            action='store',
+            dest='data_csv',
+            help='Path of csv to read enterprise learner data',
+            type=str,
+        )
+
+        parser.add_argument(
+            '-e',
+            '--enterprise_customer_uuid',
+            action='store',
+            dest='enterprise_customer_uuid',
+            help='Run this command for only the given EnterpriseCustomer UUID.'
+        )
+
+    def handle(self, *args, **options):
+        csv_path = options['data_csv']
+        enterprise_customer_uuid = options['enterprise_customer_uuid']
+        enterprise_customer = EnterpriseCustomer.objects.get(uuid=enterprise_customer_uuid)
+
+        LOGGER.info('Started Unlinking and removing DSC')
+
+        results = {}
+
+        with open(csv_path) as csv_file:
+            reader = csv.DictReader(csv_file)
+            for row in reader:
+                email = row['email']
+                results[email] = {'Unlinked': None, 'Removed_DSC': []}
+                try:
+                    EnterpriseCustomerUser.objects.unlink_user(
+                        enterprise_customer=enterprise_customer, user_email=email
+                    )
+                    results[email]['Unlinked'] = 'Success'
+                except (EnterpriseCustomerUser.DoesNotExist, PendingEnterpriseCustomerUser.DoesNotExist):
+                    message = 'Email {email} is not associated with Enterprise Customer {ec_name}'.format(
+                        email=email, ec_name=enterprise_customer.name
+                    )
+                    results[email]['Unlinked'] = message
+                    continue
+
+                user = User.objects.get(email=email)
+                enterprise_customer_user = get_enterprise_customer_user(user.id, enterprise_customer_uuid)
+                data_sharing_consent_records = enterprise_customer_user.data_sharing_consent_records
+
+                # Deleting the DCS cache
+                deleted_course_ids = []
+                for data_sharing_consent in data_sharing_consent_records:
```

Comment:
We also need to delete the actual DSC record, not just the cache (although the cache was a great catch I had forgotten about)